### DISCUSSION TITLE
Should @primer/styled-octicons be a peer dependency?

### DISCUSSION THREAD

**Discussion by colebemis** [primer/react#821] [General] [answered]
## Context

We recently created a new package called `@primer/styled-octicons` (https://github.com/primer/octicons/pull/417) that allows us to style Octicons with system props, without the need for the [`StyledOcticon`](https://primer.style/components/StyledOcticon) wrapper component. I currently have a PR (https://github.com/primer/components/pull/774) open to swap `StyledOcticon` with `@primer/styled-octicons` in Primer Components. As part of that PR, I'm trying to decide how we should handle the `@primer/styled-octicons` dependency. I've listed the pros and cons of a few options below. I'd love to hear people's thoughts on this ❤️ 

#### 🟢 Pros 

* Consumers don't have to install`@primer/styled-octicons` separately.

#### 🔴 Cons 

* Consumers might install `@primer/styled-octicons` separately if they're using Octicons in other places in their app. This could lead to a version mismatch if the version they installed is different than the version Primer Components relies on. Version mismatch could result in increased bundle size and inconsistent icons.

####  🟢 Pros 

* Consumers don't have to worry about version mismatch issues.
* Consumers can update all the Octicons in their app without having to wait for Primer Components to update its version of Octicons.

#### 🔴 Cons 

* Consumers might be confused about why they need to install the `@primer/styled-octicons` package if they don't use any components that have icons in them.
* Peer dependency errors might be frustrating and hard to understand.

#### 🟢 Pros

* Consumers don't need to install`@primer/styled-octicons` separately.

#### 🔴 Cons
* Tree shaking might break with this approach, which would result in a large increase in bundle size. 
* We'd have to release a new version of Primer components every time we ship an update to Octicons.
* Consumers might be confused about why using Octicons with Primer Components is different than using Octicons alone.

## Open questions

* Regardless of which option we choose, we should document how to use Octicons with Primer Components. Where would you expect that documentation to go?

---
cc @emplums @BinaryMuse @dmarcey

**Comment by emplums** [comment, marked as answer]
I think we should go with Option B! And it feels best to put the documentation for this in the Octicons docs, with a separate "how to use octicons with Primer Components" section in the Primer Components docs

**Reply by dmarcey** [reply]
Since there are pros/cons for each, I'm evaluating based on which option has the "least bad" con 🙃  I think that I agree with @emplums that that is B. I also like that it is consistent with what the library is doing for `styled-components`, so maybe that helps set the expectation?

**Comment by daaku** [comment]
+1 for Option B. I discovered this discussion in trying to diagnose why my build included 2 copies of octicons each weighing in at 250kb (my tree shaking setup was.. shaky). I was using v13 whereas components still depends on v11.

Maybe dependabot can help keep the dependency evergreen ;)
